### PR TITLE
Use the same redis configs in examples/guestbook-go as in examples/guestbook

### PR DESCRIPTION
**Pull request by mwielgus**
Plus updated kubectl outputs.

### CODE REVIEW

**Comment by mwielgus** [comment]
Please do not merge until Friday. Tomorrow (Thursday) there will be a workshop based on this example.

@piosz @gmarek

File: examples/guestbook-go/README.md [hunk 1]
```diff
@@ -13,38 +13,50 @@ $ hack/dev-build-and-up.sh
...
+NAME                                           READY     REASON    RESTARTS   AGE
+[...]
+redis-master-hwb4i                             1/1       Running   0          18s
 ```
 
 If you ssh to that machine, you can run `docker ps` to see the actual pod:
 
 ```shell
-me@workstation$ gcloud compute ssh --zone us-central1-b kubernetes-minion-bz1p
+me@workstation$ describe pod redis-master-hwb4i
```
Comment (piosz) [inline examples/guestbook-go/README.md]: Some part of the command is missing here.

File: examples/guestbook-go/README.md [hunk 2]
```diff
@@ -52,14 +64,16 @@ d5c458dabe50        gurpartap/redis:latest                 "/usr/local/bin/redi
...
+The pod that you created in Step One has the label `name=redis` and `role=master`. The selector field of the service determines which pods will receive the traffic sent to the service.  Use the file `examples/guestbook-go/redis-master-service.yaml` to create the service in the `kubectl` cli:
 
 ```shell
-$ kubectl create -f examples/guestbook-go/redis-master-service.json
+$ kubectl create -f examples/guestbook-go/redis-master-service.yaml
 
 $ kubectl get services
-NAME           LABELS                   SELECTOR                 IP(S)         PORT(S)
-redis-master   name=redis,role=master   name=redis,role=master   10.0.11.173   6379/TCP
+kubectl get services
```
Comment (piosz) [inline examples/guestbook-go/README.md]: duplicate of above line?

File: examples/guestbook-go/README.md [hunk 3]
```diff
@@ -67,15 +81,17 @@ This will cause all new pods to see the redis master apparently running on $REDI
...
 
 ```shell
-$ kubectl create -f examples/guestbook-go/redis-slave-controller.json
+$ kubectl create -f examples/guestbook-go/redis-slave-controller.yaml
 
 $ kubectl get rc
-CONTROLLER     CONTAINER(S)   IMAGE(S)          SELECTOR                 REPLICAS
-redis-master   redis-master   gurpartap/redis   name=redis,role=master   1
-redis-slave    redis-slave    gurpartap/redis   name=redis,role=slave    2
+kubectl get rc
```
Comment (piosz) [inline examples/guestbook-go/README.md]: Duplicate of above line?

File: examples/guestbook-go/README.md [hunk 4]
```diff
@@ -190,6 +203,8 @@ For details about limiting traffic to specific sources, see the [GCE firewall do
 
 You should delete the service which will remove any associated resources that were created e.g. load balancers, forwarding rules and target pools. All the resources (replication controllers and service) can be deleted with a single command:
 ```shell
+$ gcloud compute firewall-rules delete kubernetes-minion-3000
```
Comment (piosz) [inline examples/guestbook-go/README.md]: Please correct comment above "... with a single command".

**Comment by mwielgus** [comment]
@satnam6502  is also touching these files in #10267

**Comment by satnam6502** [comment]
Happy to fold in the conversion to YAML into my PR -- I think using YAML is much better for the example.

**Comment by satnam6502** [comment]
Or wait for my PR to merge and then adjust this PR to apply to JSON to YAML diffs.

**Comment by mwielgus** [comment]
@satnam6502 I will wait for your PR and then adjust mine.

**Comment by piosz** [comment]
LGTM from me.

@brendandburns could you please take a look as first k8s lead?

**Comment by piosz** [comment]
Ups. Brendan is OOO. @thockin could you please review or delegate?

**Comment by thockin** [comment]
LGTM

**Comment by thockin** [comment]
Needs a second LGTM if your guys's interlock is resolved

**Comment by a-robinson** [comment]
Also needs a rebase

**Comment by satnam6502** [comment]
@mwielgus  when you rebase please can you be careful not to disrupt the IP addresses etc. which I have carefully arranged to match up from the text output of commands, the screen shot etc. Perhaps it is better to strip down this PR so it only makes the switch to the YAML files and updates the links in the example from JSON to YAML? I am happy to do any/all of this.

**Comment by mwielgus** [comment]
Maybe lets reduce the scope of this PR to just using the same docker images (official + kubernetes/ instead of gurpartap/*). I will try to come up with a guestbook vs guestbook-go selectors/labels reconciliation + yaml files in a different PR.

**Comment by satnam6502** [comment]
@mwielgus scope reduction a good idea. We have a process for updating our core system Docker images -- not sure if all or some aspect of it should also apply to images for examples -- although in this case you are using existing images (I think).

**Comment by mwielgus** [comment]
@satnam6502 Yes, I took the already existing Redis images from examples/guestbook.

**Comment by satnam6502** [comment]
Great! Thanks. I guess this is ready to merge?

**Comment by mwielgus** [comment]
Yes

**Pull request merged**